### PR TITLE
Issue #2731: [UX] Manage blocks UI: No required indicator in "Admin l…

### LINKED ISSUE #2731: [UX] Manage blocks UI: No required indicator in "Admin label" field for custom blocks.
## Describe your issue or idea

While creating custom blocks from Structure -> Custom blocks -> Add custom block (`/admin/structure/block/add`) shows the "Admin label" field as required:

image

...when adding a custom block via the Manage blocks UI, it is not shown as required:

image

### Steps to reproduce (if reporting a bug)

1. Edit any layout (the Default layout for example: `admin/structure/layouts/manage/default/blocks`)
2. Add a custom block in any region by hitting the "Add block" button.
3. Click the "Make this block reusable" checkbox -> The "Admin label" field is shown, but it does not have the red asterisk that signifies a required field:

    image

4. **Without adding an admin label**, hit the "Add block" button -> Validation error saying that the "Admin label" field is required:

image

### Actual behavior (if reporting a bug)

The "Admin label" field is shown as an optional field in the add/edit custom block dialog.

### Expected behavior (if reporting a bug)

That field should have a red asterisk, indicating that it is in fact required.

---
PR upcoming...

### PR DESCRIPTION
**Pull request by klonos**
…abel" field for custom blocks.

Fixes https://github.com/backdrop/backdrop-issues/issues/2731

### CODE REVIEW

File: core/modules/layout/includes/block.text.inc [hunk 1]
```diff
@@ -76,10 +76,10 @@ class BlockText extends Block {
       $form['convert']['reusable'] = array(
         '#type' => 'checkbox',
         '#title' => t('Make this block reusable'),
-        '#description' => t('If enabled, this block will be made reusable across different layouts and be listed in on the !block_link page.', array('!block_link' => l(t('Custom blocks'), 'admin/structure/block'))),
+        '#description' => t('If enabled, this block will be made reusable across different layouts and will be listed in the !link page. This option may not be unchecked once enabled.', array('!link' => l(t('Custom blocks'), 'admin/structure/block'))),
         '#weight' => 1,
       );
-      $form['convert']['label'] = array(
+      $form['convert']['reusable-admin-label'] = array(
```
Comment (quicksketch) [inline core/modules/layout/includes/block.text.inc:82]: Hyphens are uncommon in form element "name" keys like this. Underscores should be used. But really, do we need to change these keys at all? Seems like we would minimize impact by leaving `reusable-admin-` off these keys, and it doesn't seem to affect the remainder of the changes.

**Review by quicksketch — CHANGES_REQUESTED** [review]
Changes could be minimized here, but the big issue is that tests aren't passing.

File: core/modules/layout/includes/block.text.inc (hunk 1 above)
Comment (opi) [reply core/modules/layout/includes/block.text.inc:82]: agreed

File: core/modules/layout/includes/block.text.inc [hunk 2]
```diff
@@ -89,18 +89,37 @@ class BlockText extends Block {
           'visible' => array(
             ':input[name="reusable"]' => array('checked' => TRUE),
           ),
+          // Workaround in order to make the field render as required, with a
```
Comment (opi) [inline core/modules/layout/includes/block.text.inc:92]: This is not a workaround for me, but a proper usage of #states api

File: core/modules/layout/includes/block.text.inc [hunk 3]
```diff
@@ -89,18 +89,37 @@ class BlockText extends Block {
...
+          'required' => array(
+            'input[name="reusable"]' => array('checked' => TRUE),
+          ),
         ),
       );
-      $form['convert']['delta'] = array(
+      $form['convert']['reusable-delta'] = array(
         '#type' => 'machine_name',
-        '#title' => t('Internal name'),
+        '#title' => t('Machine-readable name'),
```
Comment (opi) [inline core/modules/layout/includes/block.text.inc:103]: The word "Machine name" is used in Backdrop, we should be consistent

File: core/modules/layout/includes/block.text.inc [hunk 4]
```diff
@@ -89,18 +89,37 @@ class BlockText extends Block {
...
         '#maxlength' => 64,
         '#machine_name' => array(
-          'source' => array('convert', 'label'),
+          'source' => array('convert', 'reusable-admin-label'),
           'exists' => 'block_custom_block_load',
         ),
         '#description' => t('A unique machine-readable name containing letters, numbers, and underscores.'),
         '#weight' => 3,
+        // Workaround in order to make the field render as required, with a red
```
Comment (opi) [inline core/modules/layout/includes/block.text.inc:111]: same as before (workaround)

File: core/modules/layout/tests/layout.test [hunk 5]
```diff
@@ -1369,7 +1369,7 @@ class LayoutBlockTest extends BackdropWebTestCase {
     $this->assertText(t('Admin label is required when making a block reusable.'));
     $edit['label'] = 'My custom block label';
     $this->backdropPost(NULL, $edit, t('Update block'));
-    $this->assertText(t('An internal name is required when making a block reusable.'));
+    $this->assertText(t('A machine-readable name is required when making a block reusable.'));
```
Comment (opi) [inline core/modules/layout/tests/layout.test:1372]: same as before (machine name)

**Comment by klonos** [comment]
...I am closing this in favor of https://github.com/backdrop/backdrop/pull/2783.

Thank you @opi and @quicksketch for your review. I have incorporated the changes you suggested in the new PR.

**Pull request closed without merging**